packages/@sanity/codegen/src/typescript/findQueriesInSource.ts

```diff
@@ -51,10 +60,80 @@ export function findQueriesInSource(
           filename,
           resolver,
         })
+
         queries.push({name: queryName, result: queryResult})
       }
     },
   })
 
   return queries
 }
+
+function getDeclarationLeadingComment(
+  path: NodePath<babelTypes.VariableDeclarator>,
+): string | null {
+  /*
+   * We have to consider these cases:
+   *
+   * // @sanity-typegen-ignore
+   * const query = groq`...`
+   *
+   * // AST
+   * VariableDeclaration {
+   *   declarations: [
+   *     VariableDeclarator: {init: tag: {name: "groq"}}
+   *   ],
+   *   leadingComments: ...
+   * }
+   *
+   * // @sanity-typegen-ignore
+   * const query1 = groq`...`, query2 = groq`...`
+   *
+   * // AST
+   * VariableDeclaration {
+   *   declarations: [
+   *     VariableDeclarator: {init: tag: {name: "groq"}}
+   *     VariableDeclarator: {init: tag: {name: "groq"}}
+   *   ],
+   *   leadingComments: ...
+   * }
+   *
+   * // @sanity-typegen-ignore
+   * export const query = groq`...`
+   *
+   * // AST
+   * ExportNamedDeclaration {
+   *   declaration: VariableDeclaration {
+   *     declarations: [
+   *       VariableDeclarator: {init: tag: {name: "groq"}}
+   *       VariableDeclarator: {init: tag: {name: "groq"}}
+   *     ],
+   *   },
+   *   leadingComments: ...
+   * }
+   *
+   * In the case where multiple variables are under the same VariableDeclaration the leadingComments
+   * will still be on the VariableDeclaration
+   *
+   * In the case where the variable is exported, the leadingComments are on the
+   * ExportNamedDeclaration which includes the VariableDeclaration in its own declaration property
+   */
+
+  const variableDeclaration = path.find((node) => node.isVariableDeclaration())
+  if (!variableDeclaration) return null
+
+  if (variableDeclaration.node.leadingComments) {
+    return getLastInArray(variableDeclaration.node.leadingComments)?.value || null
+  }
+
+  // If the declaration is exported, the comment lies on the parent of the export declaration
+  if (variableDeclaration.parent.leadingComments) {
+    return getLastInArray(variableDeclaration.parent.leadingComments)?.value || null
+  }
+
+  return null
+}
+
+function getLastInArray<T>(arr: T[]) {
+  return arr[arr.length - 1]
+}
```

Comment:
Instead of just returning the last, could you also add support for any leading comments? This would match the behavior by ie `eslint-disable*`

Suggestion:
```suggestion
function declarationLeadingCommentContains(
  path: NodePath,
  comment: string
): boolean {
  /*
   * We have to consider these cases:
   *
   * // @sanity-typegen-ignore
   * const query = groq`...`
   *
   * // AST
   * VariableDeclaration {
   *   declarations: [
   *     VariableDeclarator: {init: tag: {name: "groq"}}
   *   ],
   *   leadingComments: ...
   * }
   *
   * // @sanity-typegen-ignore
   * const query1 = groq`...`, query2 = groq`...`
   *
   * // AST
   * VariableDeclaration {
   *   declarations: [
   *     VariableDeclarator: {init: tag: {name: "groq"}}
   *     VariableDeclarator: {init: tag: {name: "groq"}}
   *   ],
   *   leadingComments: ...
   * }
   *
   * // @sanity-typegen-ignore
   * export const query = groq`...`
   *
   * // AST
   * ExportNamedDeclaration {
   *   declaration: VariableDeclaration {
   *     declarations: [
   *       VariableDeclarator: {init: tag: {name: "groq"}}
   *       VariableDeclarator: {init: tag: {name: "groq"}}
   *     ],
   *   },
   *   leadingComments: ...
   * }
   *
   * In the case where multiple variables are under the same VariableDeclaration the leadingComments
   * will still be on the VariableDeclaration
   *
   * In the case where the variable is exported, the leadingComments are on the
   * ExportNamedDeclaration which includes the VariableDeclaration in its own declaration property
   */

  	const variableDeclaration = path.find((node) => node.isVariableDeclaration())
  	if (!variableDeclaration) return false
  	
  	if (variableDeclaration.node.leadingComments.some(comment => comment.value().trim() === comment)) {
  		return true
  	}
  	
  	// If the declaration is exported, the comment lies on the parent of the export declaration
  	if (variableDeclaration.parent.leadingComments.some(comment => comment.value().trim() === comment)) {
  		return true
  	}
  	
  	return false
}
```

```suggestion
function getDeclarationLeadingComment(
  path: NodePath<babelTypes.VariableDeclarator>,
): string | null {
  /*
   * We have to consider these cases:
   *
   * // @sanity-typegen-ignore
   * const query = groq`...`
   *
   * // AST
   * VariableDeclaration {
   *   declarations: [
   *     VariableDeclarator: {init: tag: {name: "groq"}}
   *   ],
   *   leadingComments: ...
   * }
   *
   * // @sanity-typegen-ignore
   * const query1 = groq`...`, query2 = groq`...`
   *
   * // AST
   * VariableDeclaration {
   *   declarations: [
   *     VariableDeclarator: {init: tag: {name: "groq"}}
   *     VariableDeclarator: {init: tag: {name: "groq"}}
   *   ],
   *   leadingComments: ...
   * }
   *
   * // @sanity-typegen-ignore
   * export const query = groq`...`
   *
   * // AST
   * ExportNamedDeclaration {
   *   declaration: VariableDeclaration {
   *     declarations: [
   *       VariableDeclarator: {init: tag: {name: "groq"}}
   *       VariableDeclarator: {init: tag: {name: "groq"}}
   *     ],
   *   },
   *   leadingComments: ...
   * }
   *
   * In the case where multiple variables are under the same VariableDeclaration the leadingComments
   * will still be on the VariableDeclaration
   *
   * In the case where the variable is exported, the leadingComments are on the
   * ExportNamedDeclaration which includes the VariableDeclaration in its own declaration property
   */

  const variableDeclaration = path.find((node) => node.isVariableDeclaration())
  if (!variableDeclaration) return null

  if (variableDeclaration.node.leadingComments) {
    return getLastInArray(variableDeclaration.node.leadingComments)?.value || null
  }

  // If the declaration is exported, the comment lies on the parent of the export declaration
  if (variableDeclaration.parent.leadingComments) {
    return getLastInArray(variableDeclaration.parent.leadingComments)?.value || null
  }

  return null
}

function getLastInArray<T>(arr: T[]) {
  return arr[arr.length - 1]
}
```